lib/actions/kb.ts

```
@@ -0,0 +1,321 @@
+
+    // Apply additional filters
+    if (validatedParams.categoryId) {
+      results = results.filter((r: any) => r.categoryId === validatedParams.categoryId);
+    }
+
+    if (validatedParams.tags && validatedParams.tags.length > 0) {
+      results = results.filter((r: any) => {
+        if (!r.tags) return false;
+        try {
+          const articleTags = JSON.parse(r.tags);
+          return validatedParams.tags!.some(tag => articleTags.includes(tag));
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)